src/services/featurehelper.js

```diff
@@ -436,15 +436,33 @@ ngeo.FeatureHelper.prototype.getVertexStyle = function(opt_incGeomFunc) {
         return;
       }
 
-      var coordinates;
+      var innerMultiCoordinates;
+      var multiCoordinates = [];
+      var coordinates = [];
+      var i, ii;
       if (geom instanceof ol.geom.LineString) {
-        coordinates = feature.getGeometry().getCoordinates();
-        return new ol.geom.MultiPoint(coordinates);
+        coordinates = geom.getCoordinates();
+      } else if (geom instanceof ol.geom.MultiLineString) {
+        multiCoordinates = geom.getCoordinates();
       } else if (geom instanceof ol.geom.Polygon) {
-        coordinates = feature.getGeometry().getCoordinates()[0];
+        coordinates = geom.getCoordinates()[0];
+      } else if (geom instanceof ol.geom.MultiPolygon) {
+        innerMultiCoordinates = geom.getCoordinates();
+      }
+
+      if (innerMultiCoordinates) {
+        for (i = 0, ii = innerMultiCoordinates.length; i < ii; i++) {
```

Comment:
I think that you don't need the `ii` variable. And a ForEach is better there no ? (same below)